### PR TITLE
update dependencies, upgrade to go 1.16.x and update goreleaser

### PR DESCRIPTION
**Pull request by cpanato**
- upgrade to Go 1.16
- update dependencies
- update goreleaser to remove deprecations


```
$ goreleaser --snapshot --skip-publish --rm-dist                                                                                                                                                                                                                                                                                                                                                                                   [67/1611]
   • releasing...
   • loading config file       file=.goreleaser.yml
   • loading environment variables
   • getting and validating git state
      • releasing v1.2.1, commit 6fd1ce2df89d0420b458d182287d72614e551f60
      • pipe skipped              error=disabled during snapshot mode
   • parsing tag
   • running before hooks
   • setting defaults
      • snapshotting
      • github/gitlab/gitea releases
      • project name
      • loading go mod information
      • building binaries
      • creating source archive
      • archives
      • linux packages
      • snapcraft packages
      • calculating checksums
      • signing artifacts
      • docker images
      • artifactory
      • blobs
      • homebrew tap formula
      • scoop manifests
      • twitter
      • milestones
   • snapshotting
   • checking ./dist
   • loading go mod information
   • writing effective config file
      • writing                   config=dist/config.yaml
   • generating changelog
      • pipe skipped              error=not available for snapshots
   • building binaries
      • building                  binary=/Users/cpanato/code/src/github.com/helm/chart-releaser/dist/chart-releaser_windows_arm_6/cr.exe
      • building                  binary=/Users/cpanato/code/src/github.com/helm/chart-releaser/dist/chart-releaser_darwin_amd64/cr
      • building                  binary=/Users/cpanato/code/src/github.com/helm/chart-releaser/dist/chart-releaser_windows_amd64/cr.exe
      • building                  binary=/Users/cpanato/code/src/github.com/helm/chart-releaser/dist/chart-releaser_linux_amd64/cr
      • building                  binary=/Users/cpanato/code/src/github.com/helm/chart-releaser/dist/chart-releaser_linux_arm_6/cr
   • archives
      • creating                  archive=dist/chart-releaser_v1.2.1-next_linux_armv6.tar.gz
      • creating                  archive=dist/chart-releaser_v1.2.1-next_linux_amd64.tar.gz
      • creating                  archive=dist/chart-releaser_v1.2.1-next_darwin_amd64.tar.gz
      • creating                  archive=dist/chart-releaser_v1.2.1-next_windows_amd64.zip
      • creating                  archive=dist/chart-releaser_v1.2.1-next_windows_armv6.zip
   • creating source archive
   • linux packages
   • snapcraft packages
   • calculating checksums
      • checksumming              file=chart-releaser_v1.2.1-next_linux_amd64.tar.gz
      • checksumming              file=chart-releaser_v1.2.1-next_linux_armv6.tar.gz
      • checksumming              file=chart-releaser_v1.2.1-next_windows_armv6.zip
      • checksumming              file=chart-releaser_v1.2.1-next_windows_amd64.zip
      • checksumming              file=chart-releaser_v1.2.1-next_darwin_amd64.tar.gz
   • signing artifacts
   • docker images
      • pipe skipped              error=publishing is disabled
   • publishing
      • blobs
      • http upload
      • custom publisher
      • artifactory
      • docker images
         • pipe skipped              error=publishing is disabled
      • docker manifests
         • pipe skipped              error=publishing is disabled
      • snapcraft packages
         • pipe skipped              error=publishing is disabled
      • github/gitlab/gitea releases
         • pipe skipped              error=publishing is disabled
      • homebrew tap formula
         • writing                   formula=dist/chart-releaser.rb
         • pipe skipped              error=publishing is disabled
      • scoop manifests
         • pipe skipped              error=publishing is disabled
      • milestones
         • pipe skipped              error=publishing is disabled
   • announcing
      • twitter
         • pipe skipped              error=announcing is disabled
   • release succeeded after 11.69s

$ docker run quay.io/helmpack/chart-releaser:latest upload --help
Upload Helm chart packages to GitHub Releases

Usage:
  cr upload [flags]

Flags:
  -c, --commit string                  Target commit for release
  -b, --git-base-url string            GitHub Base URL (only needed for private GitHub) (default "https://api.github.com/")
  -r, --git-repo string                GitHub repository
  -u, --git-upload-url string          GitHub Upload URL (only needed for private GitHub) (default "https://uploads.github.com/")
  -h, --help                           help for upload
  -o, --owner string                   GitHub username or organization
  -p, --package-path string            Path to directory with chart packages (default ".cr-release-packages")
      --release-name-template string   Go template for computing release names, using chart metadata (default "{{ .Name }}-{{ .Version }}")
      --skip-existing                  Skip upload if release exists
  -t, --token string                   GitHub Auth Token

Global Flags:
      --config string   Config file (default is $HOME/.cr.yaml)


$ /Users/cpanato/code/src/github.com/helm/chart-releaser/dist/chart-releaser_darwin_amd64/cr version
Version:         v1.2.1
Git commit:      6fd1ce2df89d0420b458d182287d72614e551f60
Date:            2021-06-28T07:33:34Z
License:         Apache 2.0

$ /Users/cpanato/code/src/github.com/helm/chart-releaser/dist/chart-releaser_darwin_amd64/cr upload --help
Upload Helm chart packages to GitHub Releases

Usage:
  cr upload [flags]

Flags:
  -c, --commit string                  Target commit for release
  -b, --git-base-url string            GitHub Base URL (only needed for private GitHub) (default "https://api.github.com/")
  -r, --git-repo string                GitHub repository
  -u, --git-upload-url string          GitHub Upload URL (only needed for private GitHub) (default "https://uploads.github.com/")
  -h, --help                           help for upload
  -o, --owner string                   GitHub username or organization
  -p, --package-path string            Path to directory with chart packages (default ".cr-release-packages")
      --release-name-template string   Go template for computing release names, using chart metadata (default "{{ .Name }}-{{ .Version }}")
      --skip-existing                  Skip upload if release exists
  -t, --token string                   GitHub Auth Token

Global Flags:
      --config string   Config file (default is $HOME/.cr.yaml)
```

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v2
         with:
-          go-version: "1.15.8"
+          go-version: "1.16.5"
```
Comment (davidkarlsen) [inline .github/workflows/ci.yaml:19]: does it make sense to use 1.16.x?

**Review by davidkarlsen — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/ci.yaml:19]: yep makes sense! thanks for the review

**Comment by davidkarlsen** [comment]
@cpanato maybe use version range on all occurrences of that action?

**Comment by cpanato** [comment]
@davidkarlsen you mean, test with go 1.15.x and 1.16.x ? or go 1.16.1 .... 1.16.5 ?

**Comment by davidkarlsen** [comment]
Use 1.16.x which gives latest 16 - but consistently anywhere setup go is used

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v2
         with:
-          go-version: "1.15.8"
+          go-version: "1.16.5"
```
Comment (davidkarlsen) [inline .github/workflows/release.yaml:22]: 1.16.x here too?

Comment (cpanato) [reply .github/workflows/release.yaml]: yep, forgot that one, thanks again!

**Pull request merged**